File: Cpp/1041_Robot_Bounded_In_Circle.cpp
class Solution {
public:
    bool isRobotBounded(string instructions) {
        int direction = 0;
        int x = 0;
        int y = 0;
        for(int i=0; i<4; ++i){
            for(auto& ch : instructions){
                if(ch == 'G') {
                    if(direction == 0) {
                        ++y;
                    }else if(direction == 1){
                        ++x;
                    }else if(direction == 2){
                        --y;
                    }else if(direction == 3){
                        --x;
                    }
                }else if(ch == 'L'){
                    direction = (direction - 1 + 4) % 4;
                }else if(ch == 'R'){
                    direction = (direction + 1) % 4;
                }
            }
        }
        return x == 0 && y == 0;
    }
};

File: Kotlin/1041_Robot_Bounded_In_Circle.kt
1041. Robot Bounded In Circle
class Solution {
    fun isRobotBounded(instructions: String): Boolean {
        var direction = 0
        var x = 0
        var y = 0
        
        for(i in 0 until 4){
            for(ch in instructions){
                when(ch) {
                    'G' -> {
                        when(direction){
                            0 -> ++y
                            1 -> ++x
                            2 -> --y
                            3 -> --x
                        }
                    }
                    'L' -> {
                        direction = (direction - 1 + 4) % 4
                    }
                    'R' ->{
                        direction = (direction + 1) % 4
                    }
                }
            }
        }
        
        return x == 0 && y == 0
    }
}